app/startup.ts

```diff
@@ -8,13 +8,15 @@ import reduxThunk from "redux-thunk";
 import { asyncSessionStorage } from "redux-persist/storages";
 import reducers from "./reducers";
 import { initRepository } from "./web3/contracts/ContractsRepository";
+import { initWeb3 } from "./web3/web3Provider";
 
-export function startup(render: (store: Store<any>) => void) {
+export async function startup(render: (store: Store<any>) => void) {
```

Comment:
this gives a false feeling like this function will return promise on completition. It won't because there is a callback at the bottom. leave a comment please or fix that callback style function to promise.